### PR TITLE
wihslist management system created

### PR DESCRIPTION
**Pull request by ivinayakg**

### CODE REVIEW

**Comment by devdeadviz** [comment]
Add a description about what you are adding in this particular pr, it helps the reviewer to get some idea of what he is getting to review.

File: src/components/ProductCard.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+          <i className={`${data.wishlist ? "fas" : "far"} fa-heart`}></i>
+        </div>
+      </div>
+      <div className="productListing_content">
+        <h5 className="productListing_title" onClick={redirectToProductPage}>
+          {data.name}
+        </h5>
+        <div className="productListing_price">
+          <h4>Rs .{data.price - (data.discount / 100) * data.price}</h4>
+          {/* <h3>Rs .{data.price}</h3> */}
```
Comment (devdeadviz) [inline src/components/ProductCard.js:46]: Removed unused code if it's not required.

**Review by devdeadviz — APPROVED** [review]
Code is clean and file structure is also good. Good work!

File: src/components/ProductCard.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        >
+          <i className={`${data.wishlist ? "fas" : "far"} fa-heart`}></i>
+        </div>
+      </div>
+      <div className="productListing_content">
+        <h5 className="productListing_title" onClick={redirectToProductPage}>
+          {data.name}
+        </h5>
+        <div className="productListing_price">
+          <h4>Rs .{data.price - (data.discount / 100) * data.price}</h4>
```
Comment (bharati-21) [inline src/components/ProductCard.js:45]: Nicely done.

File: src/components/ProductCard.js [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+        className={`productListing_action ${data.wishlist && "forWishlist"}`}
+        onClick={addToCart}
+      >
+        {!inCart
+          ? data.wishlist
+            ? "Move To Cart"
+            : "Add To Cart"
+          : "Go To Cart"}
+      </button>
+      {button && button}
```
Comment (bharati-21) [inline src/components/ProductCard.js:59]: This is such a nice way to reuse the same card for different pages and send the required buttons as props.

File: src/components/css/ProductCard.css [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+.productListing_content {
+  text-align: center;
+  background-color: white;
+  padding: 1rem;
+}
+
+.productListing_action {
+  padding: 10px;
+  background-color: rgba(var(--gray-light), 0.6);
```
Comment (bharati-21) [inline src/components/css/ProductCard.css:42]: Nice. I got to learn that you can use variables within rgba()

File: src/context & data/UserContext.js [hunk 5]
```diff
@@ -56,6 +59,21 @@ const GlobalReducer = (state, action) => {
         ...state,
         cart: cartData,
       };
+    case "wishlist":
+      if (
+        !localStorage.getItem("isAuth") ||
+        localStorage.getItem("isAuth") === "false" ||
+        localStorage.getItem("token").length < 25
+      ) {
+        notificationHandler({ type: "warning", content: "Login First..." });
```
Comment (bharati-21) [inline src/context & data/UserContext.js:68]: This is a wonderful implementation of your own Toast component.

Comment (bharati-21) [inline src/context & data/UserContext.js:68]: This is a wonderful implementation of your own Toast component.

**Review by bharati-21 — APPROVED** [review]
Well written code. I got to learn a lot from this code. Thank you for that.

Looks good to go.

**Pull request merged**